sdk/appcenter-analytics/src/main/java/com/microsoft/appcenter/analytics/Analytics.java

```diff
@@ -342,14 +342,49 @@ public static void trackEvent(String name, Map<String, String> properties) {
      * @param properties Optional properties.
      */
     public static void trackEvent(String name, EventProperties properties) {
-        trackEvent(name, properties, null);
+        trackEvent(name, properties, Flags.DEFAULT_FLAGS);
+    }
+
+    /**
+     * Track a custom event with name and optional typed properties.
+     * <p>
+     * The name cannot be null or empty.
+     * <p>
+     * The property names or values cannot be null.
+     * <p>
+     * Double values must be finite (NaN or Infinite values are discarded).
+     * <p>
+     * Additional validation rules apply depending on the configured secret.
+     * <p>
+     * For App Center:
+     * <ul>
+     * <li>The event name cannot be longer than 256 and is truncated otherwise.</li>
+     * <li>The property names cannot be empty.</li>
+     * <li>The property names and values are limited to 125 characters each (truncated).</li>
+     * <li>The number of properties per event is limited to 20 (truncated).</li>
+     * </ul>
+     * <p>
+     * For One Collector:
+     * <ul>
+     * <li>The event name needs to match the <tt>[a-zA-Z0-9]((\.(?!(\.|$)))|[_a-zA-Z0-9]){3,99}</tt> regular expression.</li>
+     * <li>The <tt>baseData</tt> and <tt>baseDataType</tt> properties are reserved and thus discarded.</li>
+     * <li>The full event size when encoded as a JSON string cannot be larger than 1.9MB.</li>
+     * </ul>
+     *
+     * @param name       An event name.
+     * @param properties Optional properties.
+     * @param flags      Optional flags. Use {@link Flags#PERSISTENCE_CRITICAL} to send this event
+     *                   before events using that use default flags or {@link Flags#PERSISTENCE_NORMAL}.
+     */
+    public static void trackEvent(String name, EventProperties properties, int flags) {
```

Comment:
Something like `@IntRange` doesn't apply here?

Reply:
We don't use them for public APIs as it triggers warnings for people not having support annotations. So we validate and log at runtime everywhere in the SDK.